web/client/components/misc/enhancers/utcDateWrapper.js

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2018, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+const { compose, withHandlers, withPropsOnChange } = require('recompose');
+const {isString, isDate} = require('lodash');
+
+const {
+    getTimezoneOffset,
+    getUTCTimeParts,
+    getUTCDateParts
+} = require('../../../utils/TimeUtils');
+
+/**
+ * Converts local date as it was UTC.
+ * Useful wrapper for calendar tools that internally use date object localized,
+ * but you want to use the date as it was UTC.
+ * @param {object} options to pass to the enhancer
+ * @param {string} options.dateType type choosed from ("date", "time", "date-time")
+ * @param {string} options.dateField date to use
+ * @param {string} options.setDateField function to use to update the date
+ * useUTCOffset is used to translate the data as it is defined in UTC.
+ * for example when using the useUTCOffset=true 20 March 2019 will correspond to 2019-03-20T00:00:00.000Z
+ * instead without it will be used as 2019-03-19T23:00:00.000Z (in cql filter), but in the UI it will be used the
+ * actual timezone
+ */
+module.exports = ({dateType = "type", dateField = 'date', setDateField = 'onSetDate'} = {}) => compose(
```

Comment:
"type" is not one of the allowed values. use `datetime`

Reply:
dateType is where to check in the props the type prop ('time', 'date', 'date-time'. 
I would call it **dateTypeProp**

Reply:
Ah, ok. Yes, fine. That's a better name